wallet/src/ui/app/pages/home/nft-details/index.tsx

```diff
@@ -21,62 +21,38 @@ import {
     useAppSelector,
     useFileExtentionType,
     useMiddleEllipsis,
+    useMediaUrl,
 } from '_hooks';
 import { accountNftsSelector } from '_redux/slices/account';
 
 import type { SuiObject } from '@mysten/sui.js';
+import type { ButtonHTMLAttributes } from 'react';
 
 import st from './NFTDetails.module.scss';
 
-function NFTDetailsPage() {
-    const [searchParams] = useSearchParams();
-    const [startNFTTransfer, setStartNFTTransfer] = useState<boolean>(false);
-    const [selectedNFT, setSelectedNFT] = useState<SuiObject | null>(null);
-    const objectId = useMemo(
-        () => searchParams.get('objectId'),
-        [searchParams]
-    );
-
-    let nftFields;
-    const nftCollections = useAppSelector(accountNftsSelector);
-
-    const activeNFT = useMemo(() => {
-        const r = nftCollections.filter(
-            (nftItems) => nftItems.reference.objectId === objectId
-        )[0];
-        setSelectedNFT(r);
-        return r;
-    }, [nftCollections, objectId]);
-
-    if (activeNFT) {
-        nftFields = isSuiMoveObject(activeNFT.data)
-            ? activeNFT.data.fields
-            : null;
-    }
-
-    const loadingBalance = useAppSelector(
-        ({ suiObjects }) => suiObjects.loading && !suiObjects.lastSync
-    );
-
-    const startNFTTransferHandler = useCallback(() => {
-        setStartNFTTransfer(true);
-    }, []);
-
-    const shortAddress = useMiddleEllipsis(nftFields?.info.id, 10, 6);
-    const fileExtentionType = useFileExtentionType(nftFields?.url || '');
-
-    if (!objectId || (!loadingBalance && !selectedNFT && !startNFTTransfer)) {
-        return <Navigate to="/nfts" replace={true} />;
-    }
-
-    const NFTDetails = nftFields && (
+function NFTdetailsContent({
+    nft,
+    onClick,
+}: {
+    nft: SuiObject;
+    onClick?: ButtonHTMLAttributes<HTMLButtonElement>['onClick'];
+}) {
+    const nftObjectID = getObjectId(nft.reference);
+    const shortAddress = useMiddleEllipsis(nftObjectID, 10, 6);
+    const filePath = useMediaUrl(nft.data);
+    const fileExtentionType = useFileExtentionType(filePath || '');
+    const nftFields = isSuiMoveObject(nft.data)
+        ? getObjectFields(nft.data)
+        : null;
+
```

Comment:
It looks like we are duplicating this logic in two files, can we add a util function that parse this information?

Reply:
Got it